src/common/common.h

```diff
@@ -86,13 +91,11 @@ extern FILE* g_error_fp;
 } while(0)
 
 #define sp_debug(debug_type, ...) do { \
-  if ((strcmp(debug_type, "injector") == 0 && getenv("SP_DEBUG_INJECTOR")) || \
-      (strcmp(debug_type, "common") == 0 && getenv("SP_DEBUG_COMMON")) || \
-      (strcmp(debug_type, "patchapi") == 0 && getenv("SP_DEBUG_PATCHAPI")) || \
-      (strcmp(debug_type, "ipc") == 0 && getenv("SP_DEBUG_IPC")) || \
-      (strcmp(debug_type, "worker") == 0 && getenv("SP_DEBUG_WORKER")) || \
-      (strcmp(debug_type, "sigtrap") == 0 && getenv("SP_DEBUG_SIGTRAP")) || \
-      (strcmp(debug_type, "agent") == 0 && getenv("SP_DEBUG_AGENT"))) { \
+  DebugType debugType = debug_type; \
+  if (debugType >= numDebugTypes) { \
+    debugType = unknownDebugType; \
+  } \
+  if (debugTypeEnabled[debugType]) { \
     if (getenv("SP_DEBUG")) {   \
```

Comment:
Put the output FILE in a variable and use it.  Then the two branches are identical. The output FILE variable can then be computed once eliminating the getenv on every sp_debug.